Cargo.toml

```diff
@@ -7,19 +7,20 @@ edition = "2021"
 [dependencies]
 chrono = "0.4.31"
 dirs = "5.0.1"
-http = "0.2.9"
-metrics = "0.21.1"
-metrics-exporter-statsd = "0.6.0"
-opentelemetry = { version = "0.20.0", features = ["rt-tokio"] }
-opentelemetry-datadog = { version = "0.8.0", features = ["reqwest-client"] }
-opentelemetry-http = "0.9.0"
+http = "0.2"
+metrics = "0.22"
+metrics-exporter-statsd = "0.7"
+opentelemetry = { version = "0.21.0" }
+opentelemetry-datadog = { version = "0.9.0", features = ["reqwest-client"] }
+opentelemetry-http = "0.10"
+opentelemetry_sdk = { version = "0.21.0", features = ["rt-tokio"] }
 serde = "1.0.189"
 serde_json = "1.0.108"
 thiserror = "1.0.50"
 tokio = "1.33.0"
 tracing = "0.1.40"
 tracing-appender = "0.2.2"
-tracing-opentelemetry = "0.21.0"
+tracing-opentelemetry = "0.22"
```

Comment:
Let stick with precise versioning to avoid any possible risk of [accidential semver breaks or crates that dont adhere to semver](https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277/9).

Reply:
When you say precise, do you mean
```
http = "0.2.9" # (which is equivalent to http = "^0.2.9")
```
or
```
http = "=0.2.9" 
```
Because the latter is the only one that is resistant to semver breaks.

Reply:
Here's the relevent command to test minimal versions
```
cargo +nightly -Z minimal-versions update
```

Reply:
Good point, `0.2.9` still can experience semver breaks. Even still, I think we should include patch versions to strictly specify a minimum version. If using `0.2`, in the event that another package uses an older version that contains a bug, our program will compile with this version.